app/app.go

```diff
@@ -966,9 +966,15 @@ func (app *App) ProcessProposalHandler(ctx sdk.Context, req *abci.RequestProcess
 				optimisticProcessingInfo.Completion <- true
 			}()
 		}
-	} else if !bytes.Equal(app.optimisticProcessingInfo.Hash, req.Hash) {
-		app.optimisticProcessingInfo.Completion <- false
+	} else {
+		app.optimisticProcessingInfo.Completion = completionSignal
```

Comment:
should we just set `app.optimisticProcessingInfo.Completion = completionSignal` in all branches? i.e. should line 959 do this too?